pkg/scheduler/plugins/reservation/plugin.go

```diff
@@ -439,6 +435,14 @@ func (pl *Plugin) filterWithReservations(ctx context.Context, cycleState *framew
 		}
 	}
 
+	if len(allInsufficientResourceReasonsByReservation) > 0 {
```

Comment:
If a pod without the reservation-affinity does not fit any matched reservation on that node, but fits the unallocated and unreserved resources of the node, it should pass the filter.